src/genie/libs/parser/iosxe/show_license.py

```diff
@@ -217,7 +217,8 @@ def cli(self, output=None):
 
         # network-advantage       (C9300-48 Network Advan...)       1 IN USE
         # dna-advantage           (C9300-48 DNA Advantage)          1 IN USE
-        p0 = re.compile(r"^(?P<license>[\w-]+)\s+\((?P<entitlement>.+)\)\s+(?P<count>\d+)\s+(?P<status>.+)$")
+        
+        p0 = re.compile(r"^(?P<license>.+\S+)\s+\((?P<entitlement>.+)\)\s+(?P<count>\d+)\s+(?P<status>.+)$")
```

Comment:
Can you provide an example of a line this will now parse that the previous one didn't? Additionally, can you update the unit tests to use that example?

Also, the `.+\S+` seems odd to me. `.+` will already capture one or more of any character. Isn't it redundant to have `\S+` after? Anything that would be captured by `\S+` will also be captured by `.+`

Reply:
I'll add an example of different output and update the unit tests with the new example.

In the new example, you'll see that there's some whitespace in the license names. The idea behind the regex was to capture all the characters in the license name (whitespace included), but not capture the whitespace after it (between the license name and entitlement tag). I'm open to a better suggestion for regex, as it probably can be done in a more efficient manner.

Reply:
I understand. My suggested solution would be to change the license capture group to this `(?P<license>.+?)`
The full line would be
```
^(?P<license>.+?)\s+\((?P<entitlement>.+)\)\s+(?P<count>\d+)\s+(?P<status>.+)$
```
The `?` after the plus will make its behavior lazy instead of greedy, so it will capture up to the whitespace, and then the \s+ regex outside the capture group will take over.

Reply:
Perfect! That's exactly what I was looking for. Thank you for the suggestion. Changes have been made to the regex pattern.